rocky/rocky/views/upload_raw.py

```diff
@@ -70,3 +80,25 @@ def process_raw(self, form):
             return self.add_error_notification(_("Raw file could not be uploaded to Bytes: %s") % str(exc))
         else:
             self.add_success_notification(_("Raw file successfully added."))
+
+    def get_form_kwargs(self):
+        kwargs = {
+            "connector": self.octopoes_api_connector,
+            "ooi_list": self.get_ooi_options(),
+        }
+        kwargs.update(super().get_form_kwargs())
+
+        if "ooi_class" in kwargs:
+            del kwargs["ooi_class"]
+
+        return kwargs
+
+    def get_ooi_options(self) -> list[tuple[str, str]]:
+        objects = self.octopoes_api_connector.list_objects(
+            set(OOI_TYPES.values()), valid_time=datetime.now(timezone.utc)
+        ).items
+
+        # generate options
+        options = [(o.primary_key, o.get_ooi_type()) for o in objects]
+
+        return options
```

Comment:
We could yield these options one at a time, to avoid building up two giant lists in memory when loading many objects.